src/i18n.tsx

```diff
@@ -1,34 +1,77 @@
 import React from 'react'
 import { i18n } from '@lingui/core'
 import { I18nProvider } from '@lingui/react'
-import { detect, fromUrl } from '@lingui/detect-locale'
-import { ReactNode, useEffect } from 'react'
-
-export const locales = ['en', 'pseudo-en', 'de', 'es-AR', 'es-US', 'it-IT', 'iw', 'ro', 'ru', 'vi', 'zh-CN', 'zh-TW']
-export const defaultLocale = 'en'
-
-const getDetectedLocale = () => {
-  const detected =
-    detect(
-      fromUrl('lang'), // helps local development
-      defaultLocale
-    ) ?? defaultLocale
-  return locales.includes(detected) ? detected : defaultLocale
+import { ReactNode } from 'react'
+import { useDispatch } from 'react-redux'
+import useParsedQueryString from 'hooks/useParsedQueryString'
+import { useEffect } from '@storybook/addons'
+import { AppDispatch } from 'state'
+import { useLocale } from 'state/user/hooks'
+
+type IE11NavigatorLanguage = {
+  userLanguage?: string
 }
 
-export async function dynamicActivate(locale: string) {
-  const { messages } = await import(`@lingui/loader!./locales/${locale}.po`)
-  i18n.loadLocaleData(locale, { plurals: () => null })
-  i18n.load(locale, messages)
-  i18n.activate(locale)
+const supportedLocales = [
+  'en',
+  'pseudo-en',
+  'de',
+  'es-AR',
+  'es-US',
+  'it-IT',
+  'iw',
+  'ro',
+  'ru',
+  'vi',
+  'zh-CN',
+  'zh-TW',
+] as const
+export type SupportedLocale = typeof supportedLocales[number]
+
+const defaultFallback = () => 'en' as SupportedLocale
+
+function parseLocale(maybeSupportedLocale: string): SupportedLocale | undefined {
+  return supportedLocales.find((locale) => locale === maybeSupportedLocale)
+}
+
+function getDetectedLocale(
+  navigator: Partial<Navigator & IE11NavigatorLanguage> = window.navigator
+): SupportedLocale | undefined {
+  const language = navigator.language ?? navigator.language
+
+  if (!language) return undefined
+
+  return parseLocale(language) ?? parseLocale(language.split('-')[0])
```

Comment:
what about upper casing the second piece? i recommend doing a split, checking length of pieces and branching instead